cocoa-mhlw/cocoa
# Debug_Mockの機能拡張（高リスク接触、陽性登録）

## Issue 番号 / Issue ID

- Close #179

## 目的 / Purpose

- #179 の実装

## 破壊的変更をもたらしますか / Does this introduce a breaking change?



```
[ ] Yes
[x] No
```

## Pull Request の種類 / Pull Request type



```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## 検証方法 / How to test

- PR #178, #189, #191 を適用し、テストします。

### コードの入手 / Get the code

```
> git clone https://github.com/cocoa-mhlw/cocoa/
> cd cocoa
cocoa> git checkout -b test
cocoa> git fetch origin pull/178/head:178
cocoa> git merge 178
cocoa> git commit -m "178"
cocoa> git fetch origin pull/189/head:189
cocoa> git merge 189
cocoa> git commit -m "189"
cocoa> git fetch origin pull/191/head:191
cocoa> git merge 191
cocoa> git commit -m "191"
cocoa> git diff --name-only HEAD..master
Covid19Radar/Covid19Radar.Android/MainApplication.cs 
Covid19Radar/Covid19Radar.Android/Services/DeviceCheckService.cs
Covid19Radar/Covid19Radar.iOS/AppDelegate.cs
Covid19Radar/Covid19Radar.iOS/Services/DeviceCheckService.cs
Covid19Radar/Covid19Radar/App.xaml.cs
Covid19Radar/Covid19Radar/Services/DeviceVerifierMock.cs
Covid19Radar/Covid19Radar/Services/ExposureNotificationHandler.cs
Covid19Radar/Covid19Radar/Services/HttpDataServiceMock.cs
Covid19Radar/Covid19Radar/Services/TestNativeImplementation.cs
Covid19Radar/Covid19Radar/ViewModels/MenuPageViewModel.cs
Covid19Radar/Covid19Radar/ViewModels/Settings/DebugPageViewModel.cs
Covid19Radar/Covid19Radar/Views/Settings/DebugPage.xaml
Covid19Radar/Covid19Radar/Views/Settings/DebugPage.xaml.cs
```

以上のファイルでOKです。（他に git 操作方法を知りませんが、もっと良い方法もあるのかもしれません）

### コードの検証 / Test the code



1. Debug_Mock ビルド、立ち上げ、初期登録、メニューからデバグページへ
2. check LastProcessTeck (long) :0
3. push FetchExposureKeyAsync & check LastProcessTeck (long) :0  意味：TEK のダウンロードは出来ていない
5. push NotifyOtherPage & input Yes, 11111 & push Register & Agree & check "Failed Please try again later. OK"   意味：陽性登録出来ない

以上が、現状と同等の挙動です。以下、新規に可能になることをチェックします。

1. settings.json を開いて下記の変更
2. "apiUrlBase": "https://API_URL_BASE/api1/r1/d1",  意味：高リスク接触発生(api1), ユーザー登録不可(r1), 陽性登録可能(d1)
3. "cdnUrlBase": "https://CDN_URL_BASE/1", 意味：毎時刻 TEK 更新(/1)
4. Debug_Mock ビルド、立ち上げ、デバグページへ
5. check CdnUrl: https://CDN_URL_BASE/1, ApiUrl: https://API_URL_BASE/api1/r1/d1 : 意味：改変 settings.json の反映
6. push FetchExposureKeyAsync & check LastProcessTeck: ~= Now, ExposureCount: (count up)  意味：毎時刻 TEK 更新(/1) & 高リスク接触発生(api1) の確認。押すたびに時刻更新＆カウンタ増加
7. push NotifyOtherPage & input Yes, 11111 & push Register & Agree & check "Complete Diagnosis Submitted OK"   意味：陽性登録可能(d1)。 PR #189 が可能にする。
8. push TutrialPage3 & push "I accept the Term of Use" & check "Network connectionerror Please ... OK"  意味：ユーザー登録不可(r1)

とりあえずは以上でOKと思われる。


## 確認事項 / What to check

- こちらは、 Android  Pixel 2 Q 10.0 API 29(Android 10.0) で確認

## その他 / Other information


- issue #169 と対応する PR #170 が、この PR に取り込まれています。
- issue #167  と対応する PR #168 が、この PR に取り込まれています。この中でユニットテストに対する言及
https://github.com/cocoa-mhlw/cocoa/pull/168#pullrequestreview-658976306
> 正常動作を確認するためにユニットテストを作成する選択肢もありますが、動作についてはコードから明らかであるので、まずは修正を取り込み、ユニットテストについては別Issueとするのが良いと考えています。

を紹介しておきます。

上記２種類のissue & PR はクローズしましたが、本当は merge すべきだったかもしれません。こういうケースにも不慣れですいません。ご指摘があれば、作業します。

----
Internal IDs:
- NFR 2581
